statistics/builder.go

```diff
@@ -184,3 +187,186 @@ func BuildColumnHist(ctx sessionctx.Context, numBuckets, id int64, collector *Sa
 func BuildColumn(ctx sessionctx.Context, numBuckets, id int64, collector *SampleCollector, tp *types.FieldType) (*Histogram, error) {
 	return BuildColumnHist(ctx, numBuckets, id, collector, tp, collector.Count, collector.FMSketch.NDV(), collector.NullCount)
 }
+
+// BuildColumnHistAndTopN build a histogram and TopN for a column from samples.
+func BuildColumnHistAndTopN(ctx sessionctx.Context, numBuckets, numTopN int, id int64, collector *SampleCollector, tp *types.FieldType,
+	count int64, ndv int64, nullCount int64) (*Histogram, *TopN, error) {
```

Comment:
How about removing arguments `count`, `ndv`, and `nullCount` since we can get them from `collector`.

Reply:
done.